ZTS build not supported.

Bug tracker for ZTS build.

/work/GIT/php-rdkafka/rdkafka.c: In function 'kafka_init':
     /work/GIT/php-rdkafka/rdkafka.c:133:69: error: 'tsrm_ls' undeclared (first use in this function)
          intern = (kafka_object*)zend_object_store_get_object(this_ptr TSRMLS_CC);
     ...                                                                     ^

--- a/metadata.c
+++ b/metadata.c
@@ -59,7 +59,7 @@ static zend_object_value kafka_metadata_new(zend_class_entry *class_type TSRMLS_
     object_intern *intern;
 
     intern = ecalloc(1, sizeof(*intern));
-    zend_object_std_init(&intern->std, class_type);
+    zend_object_std_init(&intern->std, class_type TSRMLS_CC);
     object_properties_init(&intern->std, class_type);
 
     retval.handle = zend_objects_store_put(&intern->std, (zend_objects_store_dtor_t) zend_objects_destroy_object, kafka_metadata_free, NULL TSRMLS_CC);
@@ -73,9 +73,9 @@ static zend_object_value kafka_metadata_new(zend_class_entry *class_type TSRMLS_
 }
 /* }}} */
 
-object_intern * get_object(zval *zmetadata)
+static object_intern * get_object(zval *zmetadata TSRMLS_DC)
 {
-    object_intern *ometadata = (object_intern*)zend_object_store_get_object(zmetadata);
+    object_intern *ometadata = (object_intern*)zend_object_store_get_object(zmetadata TSRMLS_CC);
 
     if (!ometadata->metadata) {
         zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Metadata::__construct() has not been called");
@@ -94,7 +94,7 @@ static HashTable *get_debug_info(zval *object, int *is_temp TSRMLS_DC) /* {{{ */
 
     array_init(&ary);
 
-    intern = get_object(object);
+    intern = get_object(object TSRMLS_CC);
     if (!intern) {
         return Z_ARRVAL(ary);
     }
@@ -122,7 +122,7 @@ PHP_METHOD(RdKafka__Metadata, getOrigBrokerId)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -145,7 +145,7 @@ PHP_METHOD(RdKafka__Metadata, getOrigBrokerName)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -168,12 +168,12 @@ PHP_METHOD(RdKafka__Metadata, getBrokers)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
 
-    kafka_metadata_collection_init(return_value, this_ptr, intern->metadata->brokers, intern->metadata->broker_cnt, sizeof(*intern->metadata->brokers), kafka_metadata_broker_ctor);
+    kafka_metadata_collection_init(return_value, this_ptr, intern->metadata->brokers, intern->metadata->broker_cnt, sizeof(*intern->metadata->brokers), kafka_metadata_broker_ctor TSRMLS_CC);
 }
 /* }}} */
 
@@ -191,12 +191,12 @@ PHP_METHOD(RdKafka__Metadata, getTopics)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
 
-    kafka_metadata_collection_init(return_value, this_ptr, intern->metadata->topics, intern->metadata->topic_cnt, sizeof(*intern->metadata->topics), kafka_metadata_topic_ctor);
+    kafka_metadata_collection_init(return_value, this_ptr, intern->metadata->topics, intern->metadata->topic_cnt, sizeof(*intern->metadata->topics), kafka_metadata_topic_ctor TSRMLS_CC);
 }
 /* }}} */
 
@@ -208,7 +208,7 @@ static const zend_function_entry kafka_metadata_fe[] = {
     PHP_FE_END
 };
 
-void kafka_metadata_minit()
+void kafka_metadata_minit(TSRMLS_D)
 {
     zend_class_entry tmpce;
 
@@ -216,13 +216,13 @@ void kafka_metadata_minit()
     ce = zend_register_internal_class(&tmpce TSRMLS_CC);
     ce->create_object = kafka_metadata_new;
 
-    kafka_metadata_topic_minit();
-    kafka_metadata_broker_minit();
-    kafka_metadata_partition_minit();
-    kafka_metadata_collection_minit();
+    kafka_metadata_topic_minit(TSRMLS_C);
+    kafka_metadata_broker_minit(TSRMLS_C);
+    kafka_metadata_partition_minit(TSRMLS_C);
+    kafka_metadata_collection_minit(TSRMLS_C);
 }
 
-void kafka_metadata_init(zval *return_value, const rd_kafka_metadata_t *metadata)
+void kafka_metadata_init(zval *return_value, const rd_kafka_metadata_t *metadata TSRMLS_DC)
 {
     object_intern *intern;
 

--- a/metadata.h
+++ b/metadata.h
@@ -16,5 +16,5 @@
   +----------------------------------------------------------------------+
 */
 
-void kafka_metadata_minit();
-void kafka_metadata_init(zval *return_value, const rd_kafka_metadata_t *metadata);
+void kafka_metadata_minit(TSRMLS_D);
+void kafka_metadata_init(zval *return_value, const rd_kafka_metadata_t *metadata TSRMLS_DC);

--- a/metadata_broker.c
+++ b/metadata_broker.c
@@ -58,7 +58,7 @@ static zend_object_value create_object(zend_class_entry *class_type TSRMLS_DC) /
     object_intern *intern;
 
     intern = ecalloc(1, sizeof(*intern));
-    zend_object_std_init(&intern->std, class_type);
+    zend_object_std_init(&intern->std, class_type TSRMLS_CC);
     object_properties_init(&intern->std, class_type);
 
     retval.handle = zend_objects_store_put(&intern->std, (zend_objects_store_dtor_t) zend_objects_destroy_object, free_object, NULL TSRMLS_CC);
@@ -72,9 +72,9 @@ static zend_object_value create_object(zend_class_entry *class_type TSRMLS_DC) /
 }
 /* }}} */
 
-static object_intern * get_object(zval *zmt)
+static object_intern * get_object(zval *zmt TSRMLS_DC)
 {
-    object_intern *omt = (object_intern*)zend_object_store_get_object(zmt);
+    object_intern *omt = (object_intern*)zend_object_store_get_object(zmt TSRMLS_CC);
 
     if (!omt->metadata_broker) {
         zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Metadata\\Broker::__construct() has not been called");
@@ -93,7 +93,7 @@ static HashTable *get_debug_info(zval *object, int *is_temp TSRMLS_DC) /* {{{ */
 
     array_init(&ary);
 
-    intern = get_object(object);
+    intern = get_object(object TSRMLS_CC);
     if (!intern) {
         return Z_ARRVAL(ary);
     }
@@ -120,7 +120,7 @@ PHP_METHOD(RdKafka__Metadata__Broker, getId)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -143,7 +143,7 @@ PHP_METHOD(RdKafka__Metadata__Broker, getHost)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -166,7 +166,7 @@ PHP_METHOD(RdKafka__Metadata__Broker, getPort)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -182,7 +182,7 @@ static const zend_function_entry fe[] = {
     PHP_FE_END
 };
 
-void kafka_metadata_broker_minit()
+void kafka_metadata_broker_minit(TSRMLS_D)
 {
     zend_class_entry tmpce;
 
@@ -191,7 +191,7 @@ void kafka_metadata_broker_minit()
     ce->create_object = create_object;
 }
 
-void kafka_metadata_broker_ctor(zval *return_value, zval *zmetadata, const void *data)
+void kafka_metadata_broker_ctor(zval *return_value, zval *zmetadata, const void *data TSRMLS_DC)
 {
     rd_kafka_metadata_broker_t *metadata_broker = (rd_kafka_metadata_broker_t*)data;
     object_intern *intern;

--- a/metadata_broker.h
+++ b/metadata_broker.h
@@ -16,5 +16,5 @@
   +----------------------------------------------------------------------+
 */
 
-void kafka_metadata_broker_minit();
-void kafka_metadata_broker_ctor(zval *return_value, zval *zmetadata, const void *metadata_broker);
+void kafka_metadata_broker_minit(TSRMLS_D);
+void kafka_metadata_broker_ctor(zval *return_value, zval *zmetadata, const void *metadata_broker TSRMLS_DC);

--- a/metadata_collection.c
+++ b/metadata_collection.c
@@ -60,7 +60,7 @@ static zend_object_value create_object(zend_class_entry *class_type TSRMLS_DC) /
     object_intern *intern;
 
     intern = ecalloc(1, sizeof(*intern));
-    zend_object_std_init(&intern->std, class_type);
+    zend_object_std_init(&intern->std, class_type TSRMLS_CC);
     object_properties_init(&intern->std, class_type);
 
     retval.handle = zend_objects_store_put(&intern->std, (zend_objects_store_dtor_t) zend_objects_destroy_object, free_object, NULL TSRMLS_CC);
@@ -70,12 +70,12 @@ static zend_object_value create_object(zend_class_entry *class_type TSRMLS_DC) /
 }
 /* }}} */
 
-static object_intern * get_object(zval *zmti)
+static object_intern * get_object(zval *zmti TSRMLS_DC)
 {
-    object_intern *omti = (object_intern*)zend_object_store_get_object(zmti);
+    object_intern *omti = (object_intern*)zend_object_store_get_object(zmti TSRMLS_CC);
 
     if (!omti->items) {
-        zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Metadata\\Collection::__construct() has not been called");
+        zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Metadata\\Collection::__construct() has not been called" TSRMLS_CC);
         return NULL;
     }
 
@@ -96,7 +96,7 @@ PHP_METHOD(RdKafka__Metadata__Collection, count)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -119,7 +119,7 @@ PHP_METHOD(RdKafka__Metadata__Collection, rewind)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -142,17 +142,17 @@ PHP_METHOD(RdKafka__Metadata__Collection, current)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
 
     if (intern->position >= intern->item_cnt) {
-        zend_throw_exception(ce_kafka_exception, "Called current() on invalid iterator", 0);
+        zend_throw_exception(ce_kafka_exception, "Called current() on invalid iterator", 0 TSRMLS_CC);
         return;
     }
 
-    intern->ctor(return_value, &intern->zmetadata, intern->items + intern->position * intern->item_size);
+    intern->ctor(return_value, &intern->zmetadata, intern->items + intern->position * intern->item_size TSRMLS_CC);
 }
 /* }}} */
 
@@ -170,13 +170,13 @@ PHP_METHOD(RdKafka__Metadata__Collection, key)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
 
     if (intern->position >= intern->item_cnt) {
-        zend_throw_exception(ce_kafka_exception, "Called key() on invalid iterator", 0);
+        zend_throw_exception(ce_kafka_exception, "Called key() on invalid iterator", 0 TSRMLS_CC);
         return;
     }
 
@@ -198,7 +198,7 @@ PHP_METHOD(RdKafka__Metadata__Collection, next)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -221,7 +221,7 @@ PHP_METHOD(RdKafka__Metadata__Collection, valid)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -240,7 +240,7 @@ static const zend_function_entry fe[] = {
     PHP_FE_END
 };
 
-void kafka_metadata_collection_minit()
+void kafka_metadata_collection_minit(TSRMLS_D)
 {
     zend_class_entry tmpce;
 
@@ -250,7 +250,7 @@ void kafka_metadata_collection_minit()
     zend_class_implements(ce TSRMLS_CC, 2, spl_ce_Countable, spl_ce_Iterator);
 }
 
-void kafka_metadata_collection_init(zval *return_value, zval *zmetadata, const void * items, size_t item_cnt, size_t item_size, kafka_metadata_collection_ctor_t ctor)
+void kafka_metadata_collection_init(zval *return_value, zval *zmetadata, const void * items, size_t item_cnt, size_t item_size, kafka_metadata_collection_ctor_t ctor TSRMLS_DC)
 {
     object_intern *intern;
 

--- a/metadata_collection.h
+++ b/metadata_collection.h
@@ -18,5 +18,5 @@
 
 typedef void (*kafka_metadata_collection_ctor_t)(zval *renurn_value, zval *zmetadata, const void *object TSRMLS_DC);
 
-void kafka_metadata_collection_minit();
-void kafka_metadata_collection_init(zval *return_value, zval *zmetadata, const void * items, size_t item_cnt, size_t item_size, kafka_metadata_collection_ctor_t ctor);
+void kafka_metadata_collection_minit(TSRMLS_D);
+void kafka_metadata_collection_init(zval *return_value, zval *zmetadata, const void * items, size_t item_cnt, size_t item_size, kafka_metadata_collection_ctor_t ctor TSRMLS_DC);

--- a/metadata_partition.c
+++ b/metadata_partition.c
@@ -59,7 +59,7 @@ static zend_object_value create_object(zend_class_entry *class_type TSRMLS_DC) /
     object_intern *intern;
 
     intern = ecalloc(1, sizeof(*intern));
-    zend_object_std_init(&intern->std, class_type);
+    zend_object_std_init(&intern->std, class_type TSRMLS_CC);
     object_properties_init(&intern->std, class_type);
 
     retval.handle = zend_objects_store_put(&intern->std, (zend_objects_store_dtor_t) zend_objects_destroy_object, free_object, NULL TSRMLS_CC);
@@ -73,9 +73,9 @@ static zend_object_value create_object(zend_class_entry *class_type TSRMLS_DC) /
 }
 /* }}} */
 
-static object_intern * get_object(zval *zmt)
+static object_intern * get_object(zval *zmt TSRMLS_DC)
 {
-    object_intern *omt = (object_intern*)zend_object_store_get_object(zmt);
+    object_intern *omt = (object_intern*)zend_object_store_get_object(zmt TSRMLS_CC);
 
     if (!omt->metadata_partition) {
         zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Metadata\\Partition::__construct() has not been called");
@@ -94,7 +94,7 @@ static HashTable *get_debug_info(zval *object, int *is_temp TSRMLS_DC) /* {{{ */
 
     array_init(&ary);
 
-    intern = get_object(object);
+    intern = get_object(object TSRMLS_CC);
     if (!intern) {
         return Z_ARRVAL(ary);
     }
@@ -123,7 +123,7 @@ PHP_METHOD(RdKafka__Metadata__Partition, getId)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -146,7 +146,7 @@ PHP_METHOD(RdKafka__Metadata__Partition, getErr)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -169,7 +169,7 @@ PHP_METHOD(RdKafka__Metadata__Partition, getLeader)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -178,7 +178,7 @@ PHP_METHOD(RdKafka__Metadata__Partition, getLeader)
 }
 /* }}} */
 
-void int32_ctor(zval *return_value, zval *zmetadata, const void *data) {
+void int32_ctor(zval *return_value, zval *zmetadata, const void *data TSRMLS_DC) {
     ZVAL_LONG(return_value, *(int32_t*)data);
 }
 
@@ -196,12 +196,12 @@ PHP_METHOD(RdKafka__Metadata__Partition, getReplicas)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
 
-    kafka_metadata_collection_init(return_value, this_ptr, intern->metadata_partition->replicas, intern->metadata_partition->replica_cnt, sizeof(*intern->metadata_partition->replicas), int32_ctor);
+    kafka_metadata_collection_init(return_value, this_ptr, intern->metadata_partition->replicas, intern->metadata_partition->replica_cnt, sizeof(*intern->metadata_partition->replicas), int32_ctor TSRMLS_CC);
 }
 /* }}} */
 
@@ -219,12 +219,12 @@ PHP_METHOD(RdKafka__Metadata__Partition, getIsrs)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
 
-    kafka_metadata_collection_init(return_value, this_ptr, intern->metadata_partition->isrs, intern->metadata_partition->isr_cnt, sizeof(*intern->metadata_partition->isrs), int32_ctor);
+    kafka_metadata_collection_init(return_value, this_ptr, intern->metadata_partition->isrs, intern->metadata_partition->isr_cnt, sizeof(*intern->metadata_partition->isrs), int32_ctor TSRMLS_CC);
 }
 /* }}} */
 
@@ -237,7 +237,7 @@ static const zend_function_entry fe[] = {
     PHP_FE_END
 };
 
-void kafka_metadata_partition_minit()
+void kafka_metadata_partition_minit(TSRMLS_D)
 {
     zend_class_entry tmpce;
 
@@ -246,7 +246,7 @@ void kafka_metadata_partition_minit()
     ce->create_object = create_object;
 }
 
-void kafka_metadata_partition_ctor(zval *return_value, zval *zmetadata, const void *data)
+void kafka_metadata_partition_ctor(zval *return_value, zval *zmetadata, const void *data TSRMLS_DC)
 {
     rd_kafka_metadata_partition_t *metadata_partition = (rd_kafka_metadata_partition_t*)data;
     object_intern *intern;

--- a/metadata_partition.h
+++ b/metadata_partition.h
@@ -16,5 +16,5 @@
   +----------------------------------------------------------------------+
 */
 
-void kafka_metadata_partition_minit();
-void kafka_metadata_partition_ctor(zval *return_value, zval *zmetadata, const void *metadata_partition);
+void kafka_metadata_partition_minit(TSRMLS_D);
+void kafka_metadata_partition_ctor(zval *return_value, zval *zmetadata, const void *metadata_partition TSRMLS_DC);

--- a/metadata_topic.c
+++ b/metadata_topic.c
@@ -60,7 +60,7 @@ static zend_object_value create_object(zend_class_entry *class_type TSRMLS_DC) /
     object_intern *intern;
 
     intern = ecalloc(1, sizeof(*intern));
-    zend_object_std_init(&intern->std, class_type);
+    zend_object_std_init(&intern->std, class_type TSRMLS_CC);
     object_properties_init(&intern->std, class_type);
 
     retval.handle = zend_objects_store_put(&intern->std, (zend_objects_store_dtor_t) zend_objects_destroy_object, free_object, NULL TSRMLS_CC);
@@ -74,9 +74,9 @@ static zend_object_value create_object(zend_class_entry *class_type TSRMLS_DC) /
 }
 /* }}} */
 
-static object_intern * get_object(zval *zmt)
+static object_intern * get_object(zval *zmt TSRMLS_DC)
 {
-    object_intern *omt = (object_intern*)zend_object_store_get_object(zmt);
+    object_intern *omt = (object_intern*)zend_object_store_get_object(zmt TSRMLS_CC);
 
     if (!omt->metadata_topic) {
         zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Metadata\\Topic::__construct() has not been called");
@@ -95,7 +95,7 @@ static HashTable *get_debug_info(zval *object, int *is_temp TSRMLS_DC) /* {{{ */
 
     array_init(&ary);
 
-    intern = get_object(object);
+    intern = get_object(object TSRMLS_CC);
     if (!intern) {
         return Z_ARRVAL(ary);
     }
@@ -122,7 +122,7 @@ PHP_METHOD(RdKafka__Metadata__Topic, getTopic)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -145,7 +145,7 @@ PHP_METHOD(RdKafka__Metadata__Topic, getErr)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -169,12 +169,12 @@ PHP_METHOD(RdKafka__Metadata__Topic, getPartitions)
         return;
     }
 
-    intern = get_object(this_ptr);
+    intern = get_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
 
-    kafka_metadata_collection_init(return_value, this_ptr, intern->metadata_topic->partitions, intern->metadata_topic->partition_cnt, sizeof(*intern->metadata_topic->partitions), kafka_metadata_partition_ctor);
+    kafka_metadata_collection_init(return_value, this_ptr, intern->metadata_topic->partitions, intern->metadata_topic->partition_cnt, sizeof(*intern->metadata_topic->partitions), kafka_metadata_partition_ctor TSRMLS_CC);
 }
 /* }}} */
 
@@ -185,7 +185,7 @@ static const zend_function_entry fe[] = {
     PHP_FE_END
 };
 
-void kafka_metadata_topic_minit()
+void kafka_metadata_topic_minit(TSRMLS_D)
 {
     zend_class_entry tmpce;
 
@@ -194,7 +194,7 @@ void kafka_metadata_topic_minit()
     ce->create_object = create_object;
 }
 
-void kafka_metadata_topic_ctor(zval *return_value, zval *zmetadata, const void *data)
+void kafka_metadata_topic_ctor(zval *return_value, zval *zmetadata, const void *data TSRMLS_DC)
 {
     rd_kafka_metadata_topic_t *metadata_topic = (rd_kafka_metadata_topic_t*)data;
     object_intern *intern;

--- a/metadata_topic.h
+++ b/metadata_topic.h
@@ -16,5 +16,5 @@
   +----------------------------------------------------------------------+
 */
 
-void kafka_metadata_topic_minit();
-void kafka_metadata_topic_ctor(zval *return_value, zval *zmetadata, const void *metadata_topic);
+void kafka_metadata_topic_minit(TSRMLS_D);
+void kafka_metadata_topic_ctor(zval *return_value, zval *zmetadata, const void *metadata_topic TSRMLS_DC);

--- a/rdkafka.c
+++ b/rdkafka.c
@@ -62,7 +62,7 @@ typedef struct _kafka_topic_object {
     zval                *zrk;
 } kafka_topic_object;
 
-static kafka_conf_object * get_kafka_conf_object(zval *zconf);
+static kafka_conf_object * get_kafka_conf_object(zval *zconf TSRMLS_DC);
 
 static const zend_function_entry empty_function_entries[] = {
     PHP_FE_END
@@ -100,7 +100,7 @@ static void kafka_free(void *object TSRMLS_DC) /* {{{ */
 }
 /* }}} */
 
-static void kafka_init(zval *this_ptr, rd_kafka_type_t type, zval *zconf, zval *zerrstr) /* {{{ */
+static void kafka_init(zval *this_ptr, rd_kafka_type_t type, zval *zconf, zval *zerrstr TSRMLS_DC) /* {{{ */
 {
     char *errstr;
     rd_kafka_t *rk;
@@ -114,7 +114,7 @@ static void kafka_init(zval *this_ptr, rd_kafka_type_t type, zval *zconf, zval *
     }
 
     if (zconf) {
-        conf_intern = get_kafka_conf_object(zconf);
+        conf_intern = get_kafka_conf_object(zconf TSRMLS_CC);
         if (conf_intern) {
             conf = rd_kafka_conf_dup(conf_intern->u.conf);
         }
@@ -142,7 +142,7 @@ static zend_object_value kafka_new(zend_class_entry *class_type TSRMLS_DC) /* {{
     kafka_object *intern;
 
     intern = ecalloc(1, sizeof(*intern));
-    zend_object_std_init(&intern->std, class_type);
+    zend_object_std_init(&intern->std, class_type TSRMLS_CC);
     object_properties_init(&intern->std, class_type);
 
     retval.handle = zend_objects_store_put(&intern->std, (zend_objects_store_dtor_t) zend_objects_destroy_object, kafka_free, NULL TSRMLS_CC);
@@ -152,12 +152,12 @@ static zend_object_value kafka_new(zend_class_entry *class_type TSRMLS_DC) /* {{
 }
 /* }}} */
 
-static kafka_object * get_kafka_object(zval *zrk)
+static kafka_object * get_kafka_object(zval *zrk TSRMLS_DC)
 {
-    kafka_object *ork = (kafka_object*)zend_object_store_get_object(zrk);
+    kafka_object *ork = (kafka_object*)zend_object_store_get_object(zrk TSRMLS_CC);
 
     if (!ork->rk) {
-        zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Kafka::__construct() has not been called");
+        zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Kafka::__construct() has not been called" TSRMLS_CC);
         return NULL;
     }
 
@@ -193,7 +193,7 @@ static zend_object_value kafka_conf_new(zend_class_entry *class_type TSRMLS_DC) 
     kafka_conf_object *intern;
 
     intern = ecalloc(1, sizeof(*intern));
-    zend_object_std_init(&intern->std, class_type);
+    zend_object_std_init(&intern->std, class_type TSRMLS_CC);
     object_properties_init(&intern->std, class_type);
 
     retval.handle = zend_objects_store_put(&intern->std, (zend_objects_store_dtor_t) zend_objects_destroy_object, kafka_conf_free, NULL TSRMLS_CC);
@@ -203,12 +203,12 @@ static zend_object_value kafka_conf_new(zend_class_entry *class_type TSRMLS_DC) 
 }
 /* }}} */
 
-static kafka_conf_object * get_kafka_conf_object(zval *zconf)
+static kafka_conf_object * get_kafka_conf_object(zval *zconf TSRMLS_DC)
 {
-    kafka_conf_object *oconf = (kafka_conf_object*)zend_object_store_get_object(zconf);
+    kafka_conf_object *oconf = (kafka_conf_object*)zend_object_store_get_object(zconf TSRMLS_CC);
 
     if (!oconf->type) {
-        zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Conf::__construct() has not been called");
+        zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Conf::__construct() has not been called" TSRMLS_CC);
         return NULL;
     }
 
@@ -235,7 +235,7 @@ static zend_object_value kafka_queue_new(zend_class_entry *class_type TSRMLS_DC)
     kafka_queue_object *intern;
 
     intern = ecalloc(1, sizeof(*intern));
-    zend_object_std_init(&intern->std, class_type);
+    zend_object_std_init(&intern->std, class_type TSRMLS_CC);
     object_properties_init(&intern->std, class_type);
 
     retval.handle = zend_objects_store_put(&intern->std, (zend_objects_store_dtor_t) zend_objects_destroy_object, kafka_queue_free, NULL TSRMLS_CC);
@@ -245,12 +245,12 @@ static zend_object_value kafka_queue_new(zend_class_entry *class_type TSRMLS_DC)
 }
 /* }}} */
 
-static kafka_queue_object * get_kafka_queue_object(zval *zrkqu)
+static kafka_queue_object * get_kafka_queue_object(zval *zrkqu TSRMLS_DC)
 {
-    kafka_queue_object *orkqu = (kafka_queue_object*)zend_object_store_get_object(zrkqu);
+    kafka_queue_object *orkqu = (kafka_queue_object*)zend_object_store_get_object(zrkqu TSRMLS_CC);
 
     if (!orkqu->rkqu) {
-        zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Queue::__construct() has not been called");
+        zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Queue::__construct() has not been called" TSRMLS_CC);
         return NULL;
     }
 
@@ -280,7 +280,7 @@ static zend_object_value kafka_topic_new(zend_class_entry *class_type TSRMLS_DC)
     kafka_topic_object *intern;
 
     intern = ecalloc(1, sizeof(*intern));
-    zend_object_std_init(&intern->std, class_type);
+    zend_object_std_init(&intern->std, class_type TSRMLS_CC);
     object_properties_init(&intern->std, class_type);
 
     retval.handle = zend_objects_store_put(&intern->std, (zend_objects_store_dtor_t) zend_objects_destroy_object, kafka_topic_free, NULL TSRMLS_CC);
@@ -290,32 +290,32 @@ static zend_object_value kafka_topic_new(zend_class_entry *class_type TSRMLS_DC)
 }
 /* }}} */
 
-static kafka_topic_object * get_kafka_topic_object(zval *zrkt)
+static kafka_topic_object * get_kafka_topic_object(zval *zrkt TSRMLS_DC)
 {
-    kafka_topic_object *orkt = (kafka_topic_object*)zend_object_store_get_object(zrkt);
+    kafka_topic_object *orkt = (kafka_topic_object*)zend_object_store_get_object(zrkt TSRMLS_CC);
 
     if (!orkt->rkt) {
-        zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Topic::__construct() has not been called");
+        zend_throw_exception_ex(NULL, 0 TSRMLS_CC, "RdKafka\\Topic::__construct() has not been called" TSRMLS_CC);
         return NULL;
     }
 
     return orkt;
 }
 
-static void new_message(zval *return_value, rd_kafka_message_t *message)
+static void new_message(zval *return_value, rd_kafka_message_t *message TSRMLS_DC)
 {
     object_init_ex(return_value, ce_kafka_message);
 
-    zend_update_property_long(NULL, return_value, ZEND_STRL("err"), message->err);
-    zend_update_property_string(NULL, return_value, ZEND_STRL("topic_name"), rd_kafka_topic_name(message->rkt));
-    zend_update_property_long(NULL, return_value, ZEND_STRL("partition"), message->partition);
+    zend_update_property_long(NULL, return_value, ZEND_STRL("err"), message->err TSRMLS_CC);
+    zend_update_property_string(NULL, return_value, ZEND_STRL("topic_name"), rd_kafka_topic_name(message->rkt) TSRMLS_CC);
+    zend_update_property_long(NULL, return_value, ZEND_STRL("partition"), message->partition TSRMLS_CC);
     if (message->payload) {
-        zend_update_property_stringl(NULL, return_value, ZEND_STRL("payload"), message->payload, message->len);
+        zend_update_property_stringl(NULL, return_value, ZEND_STRL("payload"), message->payload, message->len TSRMLS_CC);
     }
     if (message->key) {
-        zend_update_property_stringl(NULL, return_value, ZEND_STRL("key"), message->key, message->key_len);
+        zend_update_property_stringl(NULL, return_value, ZEND_STRL("key"), message->key, message->key_len TSRMLS_CC);
     }
-    zend_update_property_long(NULL, return_value, ZEND_STRL("offset"), message->offset);
+    zend_update_property_long(NULL, return_value, ZEND_STRL("offset"), message->offset TSRMLS_CC);
 }
 
 /* {{{ private constructor */
@@ -324,7 +324,7 @@ ZEND_END_ARG_INFO()
 
 PHP_METHOD(RdKafka, __construct)
 {
-    zend_throw_exception(NULL, "Private constructor", 0);
+    zend_throw_exception(NULL, "Private constructor", 0 TSRMLS_CC);
 }
 /* }}} */
 
@@ -370,7 +370,7 @@ PHP_METHOD(RdKafka__Conf, dump)
         return;
     }
 
-    intern = get_kafka_conf_object(this_ptr);
+    intern = get_kafka_conf_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -420,7 +420,7 @@ PHP_METHOD(RdKafka__Conf, set)
         return;
     }
 
-    intern = get_kafka_conf_object(this_ptr);
+    intern = get_kafka_conf_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -449,10 +449,10 @@ PHP_METHOD(RdKafka__Conf, set)
 
     switch (ret) {
         case RD_KAFKA_CONF_UNKNOWN:
-            zend_throw_exception(ce_kafka_exception, "Unknown configuration name", RD_KAFKA_CONF_UNKNOWN);
+            zend_throw_exception(ce_kafka_exception, "Unknown configuration name", RD_KAFKA_CONF_UNKNOWN TSRMLS_CC);
             break;
         case RD_KAFKA_CONF_INVALID:
-            zend_throw_exception(ce_kafka_exception, "Unknown configuration value", RD_KAFKA_CONF_INVALID);
+            zend_throw_exception(ce_kafka_exception, "Unknown configuration value", RD_KAFKA_CONF_INVALID TSRMLS_CC);
             break;
         case RD_KAFKA_CONF_OK:
             break;
@@ -491,16 +491,16 @@ PHP_METHOD(RdKafka__ConsumerTopic, consumeQueueStart)
     }
 
     if (partition != RD_KAFKA_PARTITION_UA && (partition < 0 || partition > 0x7FFFFFFF)) {
-        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Out of range value '%ld' for $partition", partition);
+        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Out of range value '%ld' for $partition", partition TSRMLS_CC);
         return;
     }
 
-    intern = get_kafka_topic_object(this_ptr);
+    intern = get_kafka_topic_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
 
-    queue_intern = get_kafka_queue_object(zrkqu);
+    queue_intern = get_kafka_queue_object(zrkqu TSRMLS_CC);
     if (!queue_intern) {
         return;
     }
@@ -509,7 +509,7 @@ PHP_METHOD(RdKafka__ConsumerTopic, consumeQueueStart)
 
     if (ret == -1) {
         err = rd_kafka_errno2err(errno);
-        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err);
+        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err TSRMLS_CC);
     }
 }
 /* }}} */
@@ -535,11 +535,11 @@ PHP_METHOD(RdKafka__ConsumerTopic, consumeStart)
     }
 
     if (partition != RD_KAFKA_PARTITION_UA && (partition < 0 || partition > 0x7FFFFFFF)) {
-        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Out of range value '%ld' for $partition", partition);
+        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Out of range value '%ld' for $partition", partition TSRMLS_CC);
         return;
     }
 
-    intern = get_kafka_topic_object(this_ptr);
+    intern = get_kafka_topic_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -548,7 +548,7 @@ PHP_METHOD(RdKafka__ConsumerTopic, consumeStart)
 
     if (ret == -1) {
         err = rd_kafka_errno2err(errno);
-        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err);
+        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err TSRMLS_CC);
     }
 }
 /* }}} */
@@ -572,11 +572,11 @@ PHP_METHOD(RdKafka__ConsumerTopic, consumeStop)
     }
 
     if (partition != RD_KAFKA_PARTITION_UA && (partition < 0 || partition > 0x7FFFFFFF)) {
-        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Out of range value '%ld' for $partition", partition);
+        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Out of range value '%ld' for $partition", partition TSRMLS_CC);
         return;
     }
 
-    intern = get_kafka_topic_object(this_ptr);
+    intern = get_kafka_topic_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -585,7 +585,7 @@ PHP_METHOD(RdKafka__ConsumerTopic, consumeStop)
 
     if (ret == -1) {
         err = rd_kafka_errno2err(errno);
-        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err);
+        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err TSRMLS_CC);
     }
 }
 /* }}} */
@@ -611,11 +611,11 @@ PHP_METHOD(RdKafka__ConsumerTopic, consume)
     }
 
     if (partition != RD_KAFKA_PARTITION_UA && (partition < 0 || partition > 0x7FFFFFFF)) {
-        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Out of range value '%ld' for $partition", partition);
+        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Out of range value '%ld' for $partition", partition TSRMLS_CC);
         return;
     }
 
-    intern = get_kafka_topic_object(this_ptr);
+    intern = get_kafka_topic_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -627,11 +627,11 @@ PHP_METHOD(RdKafka__ConsumerTopic, consume)
         if (err == RD_KAFKA_RESP_ERR__TIMED_OUT) {
             return;
         }
-        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err);
+        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err TSRMLS_CC);
         return;
     }
 
-    new_message(return_value, message);
+    new_message(return_value, message TSRMLS_CC);
 
     rd_kafka_message_destroy(message);
 }
@@ -666,7 +666,7 @@ PHP_METHOD(RdKafka__Consumer, __construct)
         return;
     }
 
-    kafka_init(this_ptr, RD_KAFKA_CONSUMER, zconf, zerrstr);
+    kafka_init(this_ptr, RD_KAFKA_CONSUMER, zconf, zerrstr TSRMLS_CC);
 
     zend_restore_error_handling(&error_handling TSRMLS_CC);
 }
@@ -694,7 +694,7 @@ PHP_METHOD(RdKafka__Kafka, addBrokers)
         return;
     }
 
-    intern = get_kafka_object(this_ptr);
+    intern = get_kafka_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -725,13 +725,13 @@ PHP_METHOD(RdKafka__Kafka, metadata)
         return;
     }
 
-    intern = get_kafka_object(this_ptr);
+    intern = get_kafka_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
 
     if (only_zrkt) {
-        only_orkt = get_kafka_topic_object(only_zrkt);
+        only_orkt = get_kafka_topic_object(only_zrkt TSRMLS_CC);
         if (!only_orkt) {
             return;
         }
@@ -740,10 +740,10 @@ PHP_METHOD(RdKafka__Kafka, metadata)
     err = rd_kafka_metadata(intern->rk, all_topics, only_orkt ? only_orkt->rkt : NULL, &metadata, timeout_ms);
 
     if (err != RD_KAFKA_RESP_ERR_NO_ERROR) {
-        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err);
+        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err TSRMLS_CC);
     }
 
-    kafka_metadata_init(return_value, metadata);
+    kafka_metadata_init(return_value, metadata TSRMLS_CC);
 }
 /* }}} */
 
@@ -763,7 +763,7 @@ PHP_METHOD(RdKafka__Kafka, setLogLevel)
         return;
     }
 
-    intern = get_kafka_object(this_ptr);
+    intern = get_kafka_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -788,7 +788,7 @@ PHP_METHOD(RdKafka__Kafka, newQueue)
         return;
     }
 
-    intern = get_kafka_object(this_ptr);
+    intern = get_kafka_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -836,13 +836,13 @@ PHP_METHOD(RdKafka__Kafka, newTopic)
         return;
     }
 
-    intern = get_kafka_object(this_ptr);
+    intern = get_kafka_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
 
     if (zconf) {
-        conf_intern = get_kafka_conf_object(zconf);
+        conf_intern = get_kafka_conf_object(zconf TSRMLS_CC);
         if (conf_intern) {
             conf = rd_kafka_topic_conf_dup(conf_intern->u.topic_conf);
         }
@@ -892,7 +892,7 @@ PHP_METHOD(RdKafka__Kafka, outqLen)
         return;
     }
 
-    intern = get_kafka_object(this_ptr);
+    intern = get_kafka_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -917,7 +917,7 @@ PHP_METHOD(RdKafka__Kafka, poll)
         return;
     }
 
-    intern = get_kafka_object(this_ptr);
+    intern = get_kafka_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -1006,22 +1006,22 @@ PHP_METHOD(RdKafka__ProducerTopic, produce)
     }
 
     if (partition != RD_KAFKA_PARTITION_UA && (partition < 0 || partition > 0x7FFFFFFF)) {
-        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Out of range value '%ld' for $partition", partition);
+        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Out of range value '%ld' for $partition", partition TSRMLS_CC);
         return;
     }
 
     if (msgflags != 0) {
-        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Invalid value '%ld' for $msgflags", msgflags);
+        zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Invalid value '%ld' for $msgflags", msgflags TSRMLS_CC);
         return;
     }
 
-    intern = get_kafka_topic_object(this_ptr);
+    intern = get_kafka_topic_object(this_ptr TSRMLS_CC);
 
     ret = rd_kafka_produce(intern->rkt, partition, msgflags | RD_KAFKA_MSG_F_COPY, payload, payload_len, key, key_len, NULL);
 
     if (ret == -1) {
         err = rd_kafka_errno2err(errno);
-        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err);
+        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err TSRMLS_CC);
     }
 }
 /* }}} */
@@ -1052,7 +1052,7 @@ PHP_METHOD(RdKafka__Producer, __construct)
         return;
     }
 
-    kafka_init(this_ptr, RD_KAFKA_PRODUCER, zconf, zerrstr);
+    kafka_init(this_ptr, RD_KAFKA_PRODUCER, zconf, zerrstr TSRMLS_CC);
 
     zend_restore_error_handling(&error_handling TSRMLS_CC);
 }
@@ -1081,7 +1081,7 @@ PHP_METHOD(RdKafka__Queue, consume)
         return;
     }
 
-    intern = get_kafka_queue_object(this_ptr);
+    intern = get_kafka_queue_object(this_ptr TSRMLS_CC);
     if (!intern) {
         return;
     }
@@ -1093,11 +1093,11 @@ PHP_METHOD(RdKafka__Queue, consume)
         if (err == RD_KAFKA_RESP_ERR__TIMED_OUT) {
             return;
         }
-        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err);
+        zend_throw_exception(ce_kafka_exception, rd_kafka_err2str(err), err TSRMLS_CC);
         return;
     }
 
-    new_message(return_value, message);
+    new_message(return_value, message TSRMLS_CC);
 
     rd_kafka_message_destroy(message);
 }
@@ -1305,12 +1305,12 @@ PHP_MINIT_FUNCTION(rdkafka)
     INIT_NS_CLASS_ENTRY(ce, "RdKafka", "Message", kafka_message_fe);
     ce_kafka_message = zend_register_internal_class(&ce TSRMLS_CC);
 
-    zend_declare_property_null(ce_kafka_message, ZEND_STRL("err"), ZEND_ACC_PUBLIC);
-    zend_declare_property_null(ce_kafka_message, ZEND_STRL("topic_name"), ZEND_ACC_PUBLIC);
-    zend_declare_property_null(ce_kafka_message, ZEND_STRL("partition"), ZEND_ACC_PUBLIC);
-    zend_declare_property_null(ce_kafka_message, ZEND_STRL("payload"), ZEND_ACC_PUBLIC);
-    zend_declare_property_null(ce_kafka_message, ZEND_STRL("key"), ZEND_ACC_PUBLIC);
-    zend_declare_property_null(ce_kafka_message, ZEND_STRL("offset"), ZEND_ACC_PUBLIC);
+    zend_declare_property_null(ce_kafka_message, ZEND_STRL("err"), ZEND_ACC_PUBLIC TSRMLS_CC);
+    zend_declare_property_null(ce_kafka_message, ZEND_STRL("topic_name"), ZEND_ACC_PUBLIC TSRMLS_CC);
+    zend_declare_property_null(ce_kafka_message, ZEND_STRL("partition"), ZEND_ACC_PUBLIC TSRMLS_CC);
+    zend_declare_property_null(ce_kafka_message, ZEND_STRL("payload"), ZEND_ACC_PUBLIC TSRMLS_CC);
+    zend_declare_property_null(ce_kafka_message, ZEND_STRL("key"), ZEND_ACC_PUBLIC TSRMLS_CC);
+    zend_declare_property_null(ce_kafka_message, ZEND_STRL("offset"), ZEND_ACC_PUBLIC TSRMLS_CC);
 
     INIT_NS_CLASS_ENTRY(ce, "RdKafka", "Queue", kafka_queue_fe);
     ce_kafka_queue = zend_register_internal_class(&ce TSRMLS_CC);
@@ -1319,7 +1319,7 @@ PHP_MINIT_FUNCTION(rdkafka)
     INIT_NS_CLASS_ENTRY(ce, "RdKafka", "Exception", NULL);
     ce_kafka_exception = zend_register_internal_class_ex(&ce, zend_exception_get_default(TSRMLS_C), NULL TSRMLS_CC);
 
-    kafka_metadata_minit();
+    kafka_metadata_minit(TSRMLS_C);
 
     return SUCCESS;
 }

--- a/rdkafka.c
+++ b/rdkafka.c
@@ -382,6 +382,8 @@ PHP_METHOD(RdKafka__Conf, dump)
         case KAFKA_TOPIC_CONF:
             dump = rd_kafka_topic_conf_dump(intern->u.topic_conf, &cntp);
             break;
+        default:
+            return;
     }
 
     array_init(return_value);
@@ -413,7 +415,7 @@ PHP_METHOD(RdKafka__Conf, set)
     int value_len;
     zval *zerrstr = NULL;
     kafka_conf_object *intern;
-    rd_kafka_conf_res_t ret;
+    rd_kafka_conf_res_t ret = 0;
     char *errstr;
 
     if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "ss|Z", &name, &name_len, &value, &value_len, &zerrstr) == FAILURE) {
@@ -861,6 +863,8 @@ PHP_METHOD(RdKafka__Kafka, newTopic)
         case RD_KAFKA_PRODUCER:
             topic_type = ce_kafka_producer_topic;
             break;
+        default:
+            return;
     }
 
     if (object_init_ex(return_value, topic_type) != SUCCESS) {

